### ISSUE TITLE
URL with "#errors" after successful password reset is misleading

### ISSUE THREAD

**Issue by brainwane** [pypa/warehouse#3168]
![warehouse-hash-errors](https://user-images.githubusercontent.com/842790/37128549-b7985d8a-2248-11e8-8942-e66b2ea4d5ba.png)
On pypi.org, I just reset my password. The green banner says "Password updated." but the URL is `https://pypi.org/manage/account/#errors`, and the `#errors` caught my eye first, confusing me. I received the confirmation email, so I know I successfully updated my password, but could we use a different word in the URL to avoid misleading the user?

**Comment by di** [comment]
Yeah, this is me being a bit lazy. Since this page has multiple separate forms on it that might cause errors, after submission we redirect to make the browser jump to any forms with errors so the message is shown to the user if it's outside the original scroll window. When there's no errors, the anchor doesn't exist, so there's no jump.

One way to fix this is to just rename the anchor to something innocuous that doesn't raise the attention of the user.

Longer-term, we might want to put these forms on separate pages, eliminating the need for this workaround -- this page is getting a little long anyways.

**Comment by shlomiLan** [comment]
Could something like [this](https://github.com/shlomiLan/warehouse/commit/8d230074d9b088aa7b078ad7c4bde5c22110d4aa) do the trick?

**Comment by di** [comment]
That would only prevent the redirect if the _current_ form has no errors, not if the submitted form has no errors.

So the first time the user loads the page, ` save_account_form.errors` would always be empty, and regardless of whether or not it was submitted with errors, it would redirect to `#`, which would break the jump to the error.

**Comment by shlomiLan** [comment]
Can you give an example? When I go to `http://localhost/manage/account/` I'm not redirected to `http://localhost/manage/account/#`.
Also, do you have an example of an error that will happen on the server side? looks like most in not all of the password logic in been done in the client side.

**Comment by di** [comment]
Looks like this isn't working 100% right now, because the `action="#errors"` is only set for the password form, but it should exist for all the forms on this page.

One example would be when a user tries to add an email address that they've already added: this generates a server-side error, which should cause the page to jump to the "Account emails" section.
